### PR TITLE
ENH: Fixing Axes functionality on the Archive Viewer

### PR DESCRIPTION
**Pull request by aksharsarvesh**
Builds on #24 

Requires the pydm PR to go through, but allows the user to cleanly navigate adding, removing, and hiding axes; as well as hiding/removing curves and it properly hiding their axes. Finally, it defaults new curves to the correct axes upon creation and allows for easy switching between axes.

### CODE REVIEW

File: archive_viewer/table_models/curve_model.py
Comment (zdomke) [inline archive_viewer/table_models/curve_model.py:118]: We should leave this conditional as is. This prevents the user from removing the last row (an empty curve) which could get the user in a stuck state.

File: archive_viewer/widgets/item_delegates.py
Comment (zdomke) [inline archive_viewer/widgets/item_delegates.py:102]: We can leave some whitespace. This will naturally break up the sections of the function and makes it easier to read.

File: archive_viewer/table_models/axis_model.py
Comment (aksharsarvesh) [reply archive_viewer/table_models/axis_model.py:84]: I posit that if somebody has axis 1, axis 2, axis 3, then deletes axis 1 and 2, a while loop would look at row count, see its 2 so attempt to create axis 2, and see that name is not being used and so will create axis 2 again. Is this good behaviour or is that not confusing

Comment (zdomke) [reply archive_viewer/table_models/axis_model.py:84]: I think the behavior you described is fine. I decided to leave it up to the user to change the axis name if they don't like the default name.

I think that your concern would make a lot of sense if the user didn't have the power to change the axis name. If it was static then I'd avoid reusing a dead name.

As for the while loop itself, it checks that the proposed name is not already in use. In the example above I simply changed "Axis 1" --> "Axis 2", then added a new axis and that broke it.

**Comment by aksharsarvesh** [comment]
Ok so after much butchering, I have taken out all of the formula dependent things about axis handling. Now, if a curve is linked or unlinked, or hidden or shown (all of which have nothing to do with formulas), then axis handling is all done in the back

File: archive_viewer/table_models/curve_model.py
Comment (YektaY) [inline archive_viewer/table_models/curve_model.py:110]:
```suggestion
        # After messing with the data, just cleanly redraw everything
```

File: archive_viewer/table_models/curve_model.py
Comment (YektaY) [inline archive_viewer/table_models/curve_model.py:136]:
```suggestion
        # By default, add a blank archivePlotCurveItem such that there's an empty row to add PVs or formulas to.
```

File: archive_viewer/table_models/curve_model.py
Comment (YektaY) [inline archive_viewer/table_models/curve_model.py:134]:
```suggestion

```

File: archive_viewer/table_models/curve_model.py
Comment (YektaY) [inline archive_viewer/table_models/curve_model.py:230]: Can you add a return type hint to the method and expand the docstring to include the input parameters

File: archive_viewer/widgets/item_delegates.py
Comment (YektaY) [inline archive_viewer/widgets/item_delegates.py:144]: can you also add the input parameters to the docstring?

File: archive_viewer/widgets/item_delegates.py
Comment (YektaY) [inline archive_viewer/widgets/item_delegates.py:322]: Same with this one, could you add to the docstring

File: archive_viewer/widgets/item_delegates.py
Comment (YektaY) [inline archive_viewer/widgets/item_delegates.py:337]: can you add to the docstring, thanks!

**Review by YektaY — APPROVED** [review]
Just a few small changes and requests. Also I see the PR has some merge conflicts. once these are fixed I will merge this in!

File: archive_viewer/widgets/item_delegates.py
Comment (aksharsarvesh) [reply archive_viewer/widgets/item_delegates.py]: Actually this was entirely on the wrong branch, so it's gone now

File: archive_viewer/table_models/curve_model.py
Comment (YektaY) [inline archive_viewer/table_models/curve_model.py:265]: this is very nit-picky of me but can you add, 

Parameters
----------

to the docstring before the 'curve: BasePlotCurveItem'

**Pull request merged**
